### ISSUE TITLE
`pod setup` does a full clone of the specs repository

### ISSUE THREAD

**Issue by Fohlen** [CocoaPods/CocoaPods#7901]
# Report

## What did you do?
I am doing a fresh workspace setup via `pod setup`

## What did you expect to happen?

It should set up the environment as fast as possible.

## What happened instead?

`/usr/local/bin/git clone https://github.com/CocoaPods/Specs.git master --progress` is issued. 
A shallow clone (e.g `--depth 1`) would be sufficient in order to get the specs running.
This way, all the old history, which is simply not necessary as an end user, is downloaded as well (that is, currently round about 550 Mb).

## CocoaPods Environment

```
   CocoaPods : 1.5.3
        Ruby : ruby 2.3.3p222 (2016-11-21 revision 56859) [universal.x86_64-darwin17]
    RubyGems : 2.5.2
        Host : Mac OS X 10.13.5 (17F77)
       Xcode :  ()
         Git : git version 2.17.0
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
HEAD
```

**Comment by dnkoutso** [comment]
Thanks for the report. This issue has been throughly discussed http://blog.cocoapods.org/Master-Spec-Repo-Rate-Limiting-Post-Mortem/ and https://github.com/CocoaPods/CocoaPods/issues/4989#issuecomment-193772935

tl;dr it is intentional and actually better not to perform a shallow clone. The initial hit is large, agreed and "killing" the master specs repo is tracked here https://github.com/CocoaPods/CocoaPods/issues/7046.

@Fohlen going to close this preemptively since there is already the issue above tracking it and it has been discussed before.
